### PR TITLE
Fix PopupMenu unresponsive on right side of menu

### LINKED ISSUE #41762: X11: Can't add effects to Audio Bus, selection impossible in popup
**Godot version:**
44e89cb88f7e982d7298e640f0828608bd3c5784

**OS/device including version:**
Mageia 8 x64, Plasma 5 on X11 with KWin

**Issue description:**
In the current `master` branch, I can't select audio effects to add to an audio bus in the popup menu with the mouse:
Screenshot_20200904_133609
I can however select items with the arrow keys and <kbd>Return</kbd>.

Possibly a regression from #41640 (CC @EricEzaM), otherwise might be a general X11 window issue (CC @pouleyKetchoupp).

**Steps to reproduce:**
- Open project on X11
- Select Audio bottom panel
- Click "Add Effect"
- Try to select an effect

**Minimal reproduction project:**
n/a

### PR DESCRIPTION
**Pull request by nathanfranke**
2020-09-01 16-20-53 mp4

*Bugsquad edit:* Fixes #41762.

### CODE REVIEW

**Comment by akien-mga** [comment]
CC @EricEzaM

**Comment by EricEzaM** [comment]
Hi @nathanfranke, @akien-mga.

Unfortunately I cannot reproduce the issue on my PR... not sure what is going on - maybe it is a platform specific issue? Anyway, this PR breaks what this piece of code tries to solve - not selecting items when trying to select the scrollbar. See below video of my test.
scrollbar_selects

So, the reason this code is required is because of the below situation. The scroll container just renders the scrollbar above the content.
popup-menu-diagram

File: scene/gui/popup_menu.cpp [hunk 1]
```diff
@@ -282,14 +282,15 @@ void PopupMenu::_gui_input(const Ref<InputEvent> &p_event) {
 	}
 
 	// Make an area which does not include v scrollbar, so that items are not activated when dragging scrollbar.
-	Rect2 item_clickable_area = control->get_global_rect();
-	float scroll_width = scroll_container->get_v_scrollbar()->is_visible_in_tree() ? scroll_container->get_v_scrollbar()->get_size().width : 0;
-	item_clickable_area.set_size(Size2(item_clickable_area.size.width - scroll_width, item_clickable_area.size.height));
+	Rect2 item_clickable_area = get_visible_rect();
```
Comment (EricEzaM) [inline scene/gui/popup_menu.cpp:285]: Option a) change `get_visible_rect()` to `scroll_container->get_rect()`. The size of the scroll container will account for the offset of the margin container, while `get_visible_rect()` gets the rect of the entire popup.

File: scene/gui/popup_menu.cpp [hunk 2]
```diff
@@ -282,14 +282,15 @@ void PopupMenu::_gui_input(const Ref<InputEvent> &p_event) {
...
+	if (scroll_container->get_v_scrollbar()->is_visible_in_tree()) {
+		item_clickable_area.size.width -= scroll_container->get_v_scrollbar()->get_size().width;
```
Comment (EricEzaM) [inline scene/gui/popup_menu.cpp:287]: Option b) also subtract get_theme_stylebox("panel").size.x to account for the width of the right hand side margin.

**Review by EricEzaM — CHANGES_REQUESTED** [review]
These changes do simplify the code quite a bit, thanks for that! I have suggested 2 options below - either work on my system. A good way to test the scrollbar is by going to the below button and changing the file dialog filter, since it has so many options.

image

**Comment by nathanfranke** [comment]
I chose option "A" since it seemed more elegant :smile:

**Comment by EricEzaM** [comment]
All good on my end after a test, thanks!

**Comment by akien-mga** [comment]
Also fixes #41762.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**